src/main/java/uk/gov/hmcts/darts/audit/api/AuditApi.java

```diff
@@ -3,13 +3,27 @@
 import uk.gov.hmcts.darts.common.entity.CourtCaseEntity;
 import uk.gov.hmcts.darts.common.entity.UserAccountEntity;
 
+import java.util.Optional;
+
 public interface AuditApi {
 
-    void record(AuditActivity activity, UserAccountEntity userAccountEntity, CourtCaseEntity courtCase);
+    default void record(AuditActivity activity, UserAccountEntity userAccountEntity, String additionalData) {
+        record(activity, userAccountEntity, Optional.empty(), Optional.of(additionalData));
+    }
+
+    default void record(AuditActivity activity, UserAccountEntity userAccountEntity) {
+        record(activity, userAccountEntity, Optional.empty(), Optional.empty());
+    }
+
+    default void record(AuditActivity activity, UserAccountEntity userAccountEntity, CourtCaseEntity courtCase) {
+        record(activity, userAccountEntity, Optional.of(courtCase), Optional.empty());
```

Comment:
Should be Optional.ofNullable(...)

Reply:
Assumed if this API call is made then the value is not null but dosent hurt